pkg/storaged/stratis-details.jsx

```diff
@@ -0,0 +1,757 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import {
+    Card, CardBody, CardTitle, CardHeader, CardActions, Text, TextVariants,
+    DescriptionList, DescriptionListTerm, DescriptionListGroup, DescriptionListDescription
+} from "@patternfly/react-core";
+import { PlusIcon, ExclamationTriangleIcon } from "@patternfly/react-icons";
+
+import { FilesystemTab, mounting_dialog, is_mounted, is_valid_mount_point, get_fstab_config } from "./fsys-tab.jsx";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { ListingPanel } from 'cockpit-components-listing-panel.jsx';
+import { StdDetailsLayout } from "./details.jsx";
+import { StorageButton, StorageLink, StorageBarMenu, StorageMenuItem, StorageUsageBar } from "./storage-controls.jsx";
+import { SidePanel, SidePanelBlockRow } from "./side-panel.jsx";
+import {
+    dialog_open,
+    TextInput, PassInput, SelectOne, SelectSpaces,
+    CheckBoxes,
+    BlockingMessage, TeardownMessage
+} from "./dialog.jsx";
+
+import {
+    fmt_size,
+    encode_filename, decode_filename,
+    get_active_usage, teardown_active_usage,
+    get_available_spaces, prepare_available_spaces,
+    reload_systemd
+} from "./utils.js";
+import { fmt_to_fragments } from "./utilsx.jsx";
+import { never_auto_explanation } from "./format-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+function teardown_block(block) {
+    return Promise.all(block.Configuration.map(c => block.RemoveConfigurationItem(c, {})));
+}
+
+function destroy_filesystem(client, fsys) {
+    const block = client.slashdevs_block[fsys.Devnode];
+    const pool = client.stratis_pools[fsys.Pool];
+
+    return teardown_block(block)
+            .then(() => {
+                return pool.call("DestroyFilesystems", [[fsys.path]])
+                        .then(([result, code, message]) => {
+                            if (code)
+                                return Promise.reject(message);
+                        });
+            });
+}
+
+function destroy_pool(client, pool) {
+    return Promise.all(client.stratis_pool_filesystems[pool.path].map(fsys => destroy_filesystem(client, fsys)))
+            .then(() => {
+                return client.stratis_manager.call("DestroyPool", [pool.path])
+                        .then(([result, code, message]) => {
+                            if (code)
+                                return Promise.reject(message);
+                        });
+            });
+}
+
+function store_passphrase(desc, passphrase) {
+    return cockpit.spawn(["stratis", "key", "set", desc, "--keyfile-path", "/dev/stdin"], { superuser: true })
+            .input(passphrase);
+}
+
+function remove_passphrase(client, key_desc) {
+    return client.stratis_manager.UnsetKey(key_desc)
+            .then((result, code, message) => {
+                if (code)
```

Comment:
Make a convenience function to handle stratisd's idiosyncratic way of returning errors.

Reply:
Ah, maybe not, this is simple enough.